build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/AbstractGradleViolationRule.groovy

```diff
@@ -42,9 +42,17 @@ import javax.inject.Inject
 abstract class AbstractGradleViolationRule extends AbstractContextAwareViolationRule {
 
     private final Map<ApiChange, String> acceptedApiChanges
+    private final File apiChangesJsonFile
+    private final File projectRootDir
 
     AbstractGradleViolationRule(Map<String, String> acceptedApiChanges) {
-        this.acceptedApiChanges = AcceptedApiChanges.fromAcceptedChangesMap(acceptedApiChanges)
+        this(acceptedApiChanges, null, null)
+    }
+
+    AbstractGradleViolationRule(Map<String, String> acceptedApiChanges, String apiChangesJsonFile, String projectRootDir) {
+        this.acceptedApiChanges = acceptedApiChanges ? AcceptedApiChanges.fromAcceptedChangesMap(acceptedApiChanges) : [:]
+        this.apiChangesJsonFile = apiChangesJsonFile ? new File(apiChangesJsonFile) : null
```

Comment:
I slightly prefer passing both paths in case we want to do anything else with either of them.  It also keeps the combination logic out the `binary-compatibility.gradle` buildscript.